File: app/src/main/java/com/example/schooldiary/mainpage/ui/MarksHolder.java
package com.example.schooldiary.mainpage.ui;

import android.view.View;
import android.widget.TextView;

import androidx.recyclerview.widget.RecyclerView;

import com.example.schooldiary.R;
import com.example.schooldiary.mainpage.model.Marks;

public class MarksHolder extends RecyclerView.ViewHolder {

    private View itemView;
    private TextView name;
    private TextView marks;

    public MarksHolder(View itemView) {
        super(itemView);
        this.itemView = itemView;
    }

    private View getItemView() {
        return itemView;
    }

    public void bindData(Marks mark) {

        name = getItemView().findViewById(R.id.subject_name);
        marks = getItemView().findViewById(R.id.subject_marks);

        name.setText(mark.getName());
        marks.setText(mark.getMarks());
    }
}

File: app/src/main/java/com/example/schooldiary/ui/main/marks/MarksHolder.kt
package com.example.schooldiary.ui.main.marks

import android.view.View
import androidx.recyclerview.widget.RecyclerView
import com.example.schooldiary.databinding.ItemMarkBinding
import com.example.schooldiary.model.marks.Marks
import kotlin.math.roundToInt

class MarksHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

    fun bindData(mark: Marks) {
        val binding = ItemMarkBinding.bind(itemView)
        binding.name.text = mark.name
        binding.allMarks.text = mark.marks
        binding.averageMark.text = calculateAverage(mark.marks)
    }

    private fun calculateAverage(str: String): String {
        var sum = 0.0
        for (element in str) {
            sum += element.toString().toDouble()
        }
        if (str.isNotEmpty()) {
            sum = (sum / str.length * 100.0).roundToInt() / 100.0
        }
        return sum.toString()
    }
}